lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,186 @@
+      -Enabled $desired_state.enabled `
+      -Description $desired_state.description `
+      -WhatIf:$check_mode
+  } Catch {
+    Fail-Json -obj $result -msg "Failed to set the AD object $($desired_state.name): $($_.Exception.Message)"`
+  }
+
+  If ($initial_state.distinguished_name -cne $desired_state.distinguished_name)
+  {
+    # Move computer to OU
+    Try {
+      Get-ADComputer -Identity $desired_state.name |
+          Move-ADObject `
+            -TargetPath $desired_state.ou `
+            -Confirm:$False `
+            -WhatIf:$check_mode
+    } Catch {
+      Fail-Json -obj $result -msg "Failed to move the AD object $($desired_state.name): $($_.Exception.Message)"`
+    }
+  }
+  $result.changed = $true
+}
+
+# ------------------------------------------------------------------------------
+Function Add-ConstructedState($desired_state) {
+  Try {
+    New-ADComputer `
+      -Name $desired_state.name `
+      -SamAccountName $desired_state.sam_account_name `
+      -DNSHostName $desired_state.dns_hostname `
+      -Path $desired_state.ou `
+      -Enabled $desired_state.enabled `
+      -Description $desired_state.description `
+      -WhatIf:$check_mode
+    } Catch {
+      Fail-Json -obj $result -msg "Failed to create the AD object $($desired_state.name): $($_.Exception.Message)"`
+    }
+
+  $result.changed = $true
+}
+
+# ------------------------------------------------------------------------------
+Function Remove-ConstructedState($initial_state) {
+  Try {
+    Remove-ADComputer `
+      -Identity $initial_state.name `
+      -Confirm:$False `
+      -WhatIf:$check_mode
+  } Catch {
+    Fail-Json -obj $result -msg "Failed to remove the AD object $($desired_state.name): $($_.Exception.Message)"`
+  }
+
+  $result.changed = $true
+}
+
+# ------------------------------------------------------------------------------
+Function are_hashtables_equal($x, $y) {
+  # Compare not nested HashTables
+  Foreach ($key in $x.Keys) {
+      If (($y.Keys -notcontains $key) -or ($x[$key] -cne $y[$key])) {
+          Return $false
+      }
+  }
+  foreach ($key in $y.Keys) {
+      if (($x.Keys -notcontains $key) -or ($x[$key] -cne $y[$key])) {
+          Return $false
+      }
+  }
+  Return $true
+}
+
+# ------------------------------------------------------------------------------
+$initial_state = Get-InitialState($desired_state)
+
+If ($desired_state.state -eq "present") {
+    If ($initial_state.state -eq "present") {
+      $in_desired_state = are_hashtables_equal $initial_state $desired_state
+
+      If (-not $in_desired_state) {
+        Set-ConstructedState $initial_state $desired_state
+      }
+    } Else { # $desired_state.state = "Present" & $initial_state.state = "Absent"
+      Add-ConstructedState($desired_state)
+    }
+  } Else { # $desired_state.state = "Absent"
+    If ($initial_state.state -eq "present") {
+      Remove-ConstructedState($initial_state)
+    }
+  }
+
+If ($diff_support) {
```

Comment:
Does putting a hashtable in before and after actually work, I've never tried it before so not sure.

Reply:
It works OK, and I think that makes diff implementation easy.

Reply:
Thanks, I never knew this and it definitely makes things a lot easier.